b-real-time-htb.js

```diff
@@ -263,15 +285,17 @@ function BRealTimeHtb(configs) {
 
             /* ---------- Fill the bid variables with data from the bid response here. ------------*/
 
-            var bidPrice; // the bid price for the given slot
-            var bidWidth; // the width of the given slot
-            var bidHeight; // the height of the given slot
-            var bidCreative; // the creative/adm for the given slot that will be rendered if is the winner.
-            var bidDealId; // the dealId if applicable for this slot.
-            var bidIsPass; // true/false value for if the module returned a pass for this slot.
+            var bidIsPass = bid.nobid; // true/false value for if the module returned a pass for this slot.
+            if(!bidIsPass) {
+              var bidPrice = bid.ads[0].cpm; // the bid price for the given slot
+              var bidWidth = bid.ads[0].rtb.banner.width; // the width of the given slot
+              var bidHeight = bid.ads[0].rtb.banner.height; // the height of the given slot
+              var bidCreative = bid.ads[0].rtb.banner.content; // the creative/adm for the given slot that will be rendered if is the winner.
+              var bidDealId = (typeof bid.ads[0].deal_id !== 'undefined') ? bid.ads[0].deal_id : null; // the dealId if applicable for this slot.
+            }
 
             /* ---------------------------------------------------------------------------------------*/
-
+            
             if (bidIsPass) {
                 //? if (DEBUG) {
                 Scribe.info(__profile.partnerId + ' returned pass for { id: ' + adResponse.id + ' }.');
```

Comment:
Can you please add `requestId: curReturnParcel.requestId` after line 308 here https://github.com/indexexchange/b-real-time-adapter/blob/83305a3ec0833d93de91db670cab2b78b08d4aa7/b-real-time-htb.js#L304-L309 

```
EventsService.emit('hs_slot_pass', {
    sessionId: sessionId,
    statsId: __profile.statsId,
    htSlotId: curReturnParcel.htSlot.getId(),
    xSlotNames: [curReturnParcel.xSlotName],
    requestId: curReturnParcel.requestId
});
 ```

Reply:
In addition, after line 322 here
https://github.com/indexexchange/b-real-time-adapter/blob/83305a3ec0833d93de91db670cab2b78b08d4aa7/b-real-time-htb.js#L318-L323 

```
EventsService.emit('hs_slot_bid', {
    sessionId: sessionId,
    statsId: __profile.statsId,
    htSlotId: curReturnParcel.htSlot.getId(),
    xSlotNames: [curReturnParcel.xSlotName],
    requestId: curReturnParcel.requestId
});
```